### ISSUE TITLE
Atom cannot install atom-beautify 2018/08/31

### ISSUE THREAD

**Issue by francosmp** [Glavin001/atom-beautify#2213]
# Description

The results of beautification are not what I expect.

# Input Before Beautification

This is what the code looked like before:

```
apm install atom-beautify
```

# Expected Output

The beautified code should have looked like this:

```
//Installation
```

# Actual Output

The beautified code actually looked like this:

```
npm WARN deprecated graceful-fs@2.0.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated @types/commander@2.12.2: This is a stub types definition for commander (https://github.com/tj/commander.js). commander provides its own type definitions, so you don't need @types/commander installed!
npm WARN deprecated jscs-preset-wikimedia@1.0.1: Deprecated in favour of eslint-config-wikimedia. -- https://phabricator.wikimedia.org/T118941
npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
npm ERR! Windows_NT 10.0.17134
npm ERR! argv "C:\\Users\\iFranco\\AppData\\Local\\atom\\app-1.30.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\iFranco\\AppData\\Local\\atom\\app-1.30.0\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\iFranco\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\iFranco\\.atom\\.apmrc" "install" "C:\\Users\\iFranco\\AppData\\Local\\Temp\\d-118731-7472-1rzyzv6.ry0uj714i\\package.tgz" "--runtime=electron" "--target=2.0.5" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code EREADFILE

npm ERR! Error extracting C:\Users\iFranco\.atom\.apm\marko\4.13.3\package.tgz archive: ENOENT: no such file or directory, open 'C:\Users\iFranco\.atom\.apm\marko\4.13.3\package.tgz'
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\iFranco\AppData\Local\Temp\apm-install-dir-118731-7472-1n6hv2y.ak2xz5b3xr\npm-debug.log
```

# Steps to Reproduce

1. Add code to Atom editor
2. Run command `Atom Beautify: Beautify Editor`
3. This beautified code does not look right!

# Debug

Here is a link to the `debug.md` Gist: <INSERT GIST HERE>

# Checklist

I have:
- [x] Tried uninstalling and reinstalling Atom Beautify to ensure it installed properly
- [x] Reloaded (or restarted) Atom to ensure it is not a caching issue
- [x] Searched through existing Atom Beautify Issues at https://github.com/Glavin001/atom-beautify/issues
so I know this is not a duplicate issue
- [x] Filled out the Input, Expected, and Actual sections above or have edited/removed them in a way that fully describes the issue.
- [ ] Generated debugging information by executing `Atom Beautify: Help Debug Editor` command in Atom and added link for `debug.md` Gist to this issue

**Comment by TonyApuzzo** [comment]
bad bot.

Can't issue atom beautify command if you can't install the package!

**Comment by TonyApuzzo** [comment]
@francosmp guessing if you just create a junk debug.md file upload gist to link that will mollify the overzealous bot.

**Comment by szeck87** [comment]
Ignore probot-no-response.  Do you have the `language-marko` package installed as well?

**Comment by TonyApuzzo** [comment]
(I'm not the OP, but seemingly have the same problem.)  I can't install language-marko standalone either:

```
[fdidev@ia16ep4demo ~]$ apm clean
[fdidev@ia16ep4demo ~]$ apm install language-marko
Installing language-marko to /home/fdidev/.atom/packages ✗

npm ERR! Linux 3.10.0-862.11.6.el7.x86_64
npm ERR! argv "/usr/share/atom/resources/app/apm/bin/node" "/usr/share/atom/resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/home/fdidev/.atom/.apm/.apmrc" "--userconfig" "/home/fdidev/.atom/.apmrc" "install" "/tmp/d-118731-10770-wtbmjg.b2ydkakyb9/package.tgz" "--runtime=electron" "--target=2.0.5" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code EREADFILE

npm ERR! Error extracting /home/fdidev/.atom/.apm/marko/4.13.3/package.tgz archive: ENOENT: no such file or directory, open '/home/fdidev/.atom/.apm/marko/4.13.3/package.tgz'
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /tmp/apm-install-dir-118731-10770-tcxall.t0bdpfogvi/npm-debug.log

[fdidev@ia16ep4demo ~]$
```

The apm log doesn't seem to exist either:

```
[fdidev@ia16ep4demo ~]$ cat /tmp/apm-install-dir-118731-10770-tcxall.t0bdpfogvi/npm-debug.log
cat: /tmp/apm-install-dir-118731-10770-tcxall.t0bdpfogvi/npm-debug.log: No such file or directory
[fdidev@ia16ep4demo ~]$
```

**Comment by szeck87** [comment]
Are you trying to install `language-marko` or `atom-beautify`?  Or both?

**Comment by TonyApuzzo** [comment]
Well, first just `atom-beautify` but then `language-marko` by itself.  Guessing (wild guess) that is really the core problem and not particularly `atom-beautify`.  Strangely it seems to be trying to install `language-marko` at a version that doesn't exist.  4.13.3 is not a version of `language-marko`.  I tried forcing it with `language-marko@2.9.4` and it failed with same error.

Also tried completely uninstalling `sudo yum erase atom` and removing `~/.atom`, then reinstalling.  No joy.

So probably I (or OP) should go over to `language-marko` and put in a bug report there.

**Comment by TonyApuzzo** [comment]
I'm not an atom user, just a hapless devops guy trying to fix an ansible script that started failing all of a sudden doing `apm install atom-beautify`.  So this will go back into a queue for someone who actually knows atom to investigate further on our side.

**Comment by szeck87** [comment]
OK, I'm getting the same error on my Mac now.

It's actually the core [marko](https://github.com/marko-js/marko) package that is failing to install here, which is required by [marko-prettyprint](https://github.com/marko-js/marko-prettyprint), which is required by atom-beautify.  4.13.3 is a valid version of that, but it looks like it's actually downloading 4.13.4.

Unfortunately that debug log disappears after failure, so can't even look at it.

**Comment by francosmp** [comment]
well, it was great until i had problems with maximizing atom. so i uninstalled it. after uninstall, i deleted atom in program files, program files x86, program data, users, and users app data. then i downloaded atom from oficial page and i installed it. fillany i installed many packages but the problem was with beautify. i tried install with setting -> install and got the same problem that console install.
what's next :(

**Comment by szeck87** [comment]
@Glavin001 might have to remove `marko-prettyprint` as a dependency and do a release.

**Comment by daneelgiskard** [comment]
Same issue here - looking foward to a solution. Many thanks for your efforts.

**Comment by default50** [comment]
Just opened [this](https://github.com/marko-js/atom-language-marko/issues/26) new issue agains language-marki as I believ it's the root cause.

For **atom-beautify**, creating a symling works around the installation issue:

```
~/.atom/.apm $ ll marko
total 8.0K
drwxrwxr-x 3 crux crux 4.0K May 26 21:43 4.10.0
lrwxrwxrwx 1 crux crux    6 Sep  1 16:59 4.13.3 -> 4.13.4
drwxr-xr-x 3 crux crux 4.0K Sep  1 16:51 4.13.4
```

**Comment by szeck87** [comment]
It's not language-marko causing the error in atom-beautify.  They just share the same core issue.

**Comment by 0xYasser** [comment]
I've the same issue .. I get the same error when I try ```apm install atom-beautify```

**Comment by kmgumienny** [comment]
Also getting this same issue. I tried updating the package but that failed, so I uninstalled and when I tried to reinstall I couldn't. Waiting on a fix.

**Comment by yingxuegu** [comment]
I have the same issue for both mac and window. I tried apm, but it does not work either.

**Comment by lgharapetian** [comment]
THANK YOU! @default50 your solution solves it: https://github.com/Glavin001/atom-beautify/issues/2213#issuecomment-417869719

Recap:
cd /Users/<userID>/.atom/.apm/marko;  ln -s 4.13.4 4.13.3; apm install atom-beautify

**Comment by yingxuegu** [comment]
@default50 thank you! but how to fix it on windows?

**Comment by yingxuegu** [comment]
I solve it on windows with git bash and @default50 #2213

**Comment by ar012** [comment]
Many many thanks.
On Linux, the issue is solved by taking the following steps:
cd /home/your_username/.atom/.apm/marko
ln -s 4.13.4 4.13.3
apm install atom-beautify

**Comment by TBA-projects** [comment]
on windows use comand mklink ... like
"mklink /J c:\Users\username\.atom\.apm\marko\4.13.3 c:\Users\user\.atom\.apm\marko\4.13.4"

**Comment by leandrofecr** [comment]
@TBA-projects , How do I do this?


> on windows use comand mklink ... like
> mklink /J c:\Users\username\.atom\.apm\marko\4.13.3 c:\Users\username\.atom\.apm\marko\4.13.4

**Comment by malopriso** [comment]
Installing “atom-beautify@0.33.0” failed.Hide output…

npm WARN deprecated graceful-fs@2.0.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated @types/commander@2.12.2: This is a stub types definition for commander (https://github.com/tj/commander.js). commander provides its own type definitions, so you don't need @types/commander installed!
npm WARN deprecated jscs-preset-wikimedia@1.0.1: Deprecated in favour of eslint-config-wikimedia. -- https://phabricator.wikimedia.org/T118941
npm WARN deprecated nomnom@1.8.1: Package no longer supported. Contact support@npmjs.com for more info.
npm ERR! Darwin 17.7.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/user/.atom/.apm/.apmrc" "--userconfig" "/Users/user/.atom/.apmrc" "install" "/private/var/folders/fy/ff39dmw95cx6ks3npnpkk2880000gn/T/d-11883-7509-1niq8xu.dkc87x2yb9/package.tgz" "--runtime=electron" "--target=2.0.5" "--arch=x64" "--global-style"
npm ERR! node v6.9.5
npm ERR! npm  v3.10.10
npm ERR! code EREADFILE

npm ERR! Error extracting /Users/user/.atom/.apm/marko/4.13.3/package.tgz archive: ENOENT: no such file or directory, open '/Users/user/.atom/.apm/marko/4.13.3/package.tgz'
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /private/var/folders/fy/ff39dmw95cx6ks3npnpkk2880000gn/T/apm-install-dir-11883-7509-1e8lcyb.mmeku766r/npm-debug.log


same error

**Comment by szeck87** [comment]
I just published v0.33.1.  For the time being, marko-prettyprint has been removed as there is an issue with the marko dependency when downloading.

**Comment by fonkamloic** [comment]
**For Linux Users**

After the installation fails, it will show you the version which failed. As of today, the latest version is `atom-beautify@0.33.4` What you should do is to navigate to atom-beautify page, click on the versions tab, download the previous version which for me is `atom-beautify@0.33.3` You can download it as a .zip then you should do the following:

```
cd /home/yourUsername/.atom/packages
unzip atom-beautify.zip
mv atom-beautify-0.33.3 atom-beautify
```
Then restart your atom editor, and I should be installed already. If not click install again and it will refresh
